projects/ngrx.io/content/guide/data/entity-metadata.md

```diff
@@ -249,6 +250,7 @@ export class AdditionalPropertyPersistenceResultHandler extends DefaultPersisten
 Following the prior step, we have added the additional property to the `action.payload`. Up next we need to set it to the instance of EntityCollection in the `reducer`. In order to accomplish that, we need to create an `AdditionalEntityCollectionReducerMethods` that `extends EntityCollectionReducerMethods`. In addition, we will need to overwrite the method to match your `action`. For example, if the additional property `foo` is only available in `queryMany action(triggered by EntityCollectionService.getWithQuery)`, we can follow this approach.
 
 ```typescript
+@Injectable()
 export class AdditionalEntityCollectionReducerMethods<T> extends EntityCollectionReducerMethods<T> {
```

Comment:
Not sure how this class got here (did I write it?).   It's not quite correct as it it mutates the collection (`ec`). That's OK if `ec` is new but, under some circumstances, it might be the original collection.  I'd want to look at that. I'd also like to get rid of the duplicative `any` "casting".  I'll weigh in soon.

Reply:
No, that was created by @JiaLiPassion
>Not sure how this class got here (did I write it?)